cmd/soroban-rpc/internal/integrationtest/infrastructure/test.go

```diff
@@ -170,6 +174,10 @@ func NewTest(t *testing.T, cfg *TestConfig) *Test {
 		i.waitForRPC()
 	}
 
+	// We populate the transactions table with some rows so that GetLedgerRange called in transaction related
+	// integration tests returns a valid latest ledger.
+	i.insertTransactions()
```

Comment:
Please don't do this for every test, this is related to transaction-tests only. Also, why can't you submit transactions to populate the table instead of doing it artificially?

Reply:
Well, because I added this in the first place was because the integration tests for sending transaction were failing here - https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/integrationtest/transaction_test.go#L179. I thought it would defeat the purpose of testing the send functionality if we use it to populate the table artificially. What do you think?

However, I agree that we should not run these on each integration test and I will add a config to do it only for txn related tests.

Reply:
If that code is only used in getTransaction tests, then we should try to only incorporate it there.

I am not sure why existing tests were failing. It worries me that we need to artificially need to populate the db to fix them.

Reply:
@2opremio No its not just used in getTransactions tests. There were around 11 integration tests that were failing due to not populating the the transactions db. So this piece of code is tied to multiple tests.

Reply:
> There were around 11 integration tests that were failing due to not populating the the transactions db

This is the part I don't understand. The normal operation of tests should populate the transactions table.

Reply:
i.e. if you want the transactions table to be populated, then submit transactions. If you need to artificially fill the table for some tests, then those tests should do it and the general `test.go` file shouldn't be poluted by it.